### ISSUE TITLE
Accessibility: GridField adds a header for actions even if there's no actions column

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-framework#11817]
[WCAG 2.2 A 1.3.1](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
Tables shouldn't have header cells that don't have corresponding data cells.

If no rows in a gridfield have actions, the gridfield still creates a `th` that looks like:
`<th class='main col-Actions'>`

## Acceptance Criteria

- [x] If no rows have actions, there isn't a `th` for actions
- [ ] If even a single row has actions, there's a corresponding `th` for actions
- [ ] If even a single row has actions, all rows have a cell for actions (even if there's no actions on that specific row)

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11848
- https://github.com/silverstripe/silverstripe-framework/pull/11882
- https://github.com/silverstripe/silverstripe-versioned/pull/529
- https://github.com/silverstripe/silverstripe-admin/pull/2020
- https://github.com/silverstripe/silverstripe-gridfieldextensions/pull/460
- https://github.com/silverstripe/developer-docs/pull/830

**Comment by dcentrica** [comment]
+1 for this. I needed to workaround the missing "Actions" header just yesterday.

**Comment by GuySartorelli** [comment]
@dcentrica If the "Actions" table header is _actually missing_ that's a separate bug which may not be resolved by this issue. Please open a separate issue for that. 

This issue is mostly about the fact that the "Actions" table header is present even when it shouldn't be.
